Combine Steering and TO Committees (#1587)
knative/community: STEERING-COMMITTEE.md

> | <img width="30px" src="https://github.com/evankanderson.png"> | Evan Anderson           | VMware       | [@evankanderson](https://github.com/evankanderson) | 2023-11-28 | 2025     |

Reviewer: unrelated, but @evankanderson do you want to update your company affiliation too?

Author: Oops, thank you!

Author: In fact, I wasn't working at VMware when I was elected...

## Getting in touch

There are three ways to raise issues to the steering committee for decision:

1. Emailing the steering committee at
   [steering@knative.team](mailto:steering@knative.team).
   This is a private discussion list to which all members of the committee have
   access.
2. Open an issue on a knative repository and indicate that you would like
   attention from the steering committee.
   
3. Asking a question on the
   [#knative-toc-sc-questions](https://cloud-native.slack.com/archives/C04LQCW0C03)
   channel on [CNCF Slack](https://cloud-native.slack.com/).

## Composition

> The Steering Committee has five (7) seats. These 7 seats are composed as follows:

Reviewer 2: ```suggestion
The Steering Committee has seven (7) seats. These 7 seats are composed as follows:
```

> The Steering Committee has five (7) seats. These 7 seats are composed as follows:

Reviewer 3: ```suggestion
The Steering Committee has a minimum of 5 (five) and maximum of (7) seats. These 7 seats are composed as follows:
```

Reviewer 3: Since selected seats are optional, I've made a suggestion to update the text here. We also need to note that these are optional.

Maybe me writing `optional` is bad here. We should encourage SC to fill in those seats.

Author: I actually think those seats are _optional_.  If we ended up with 5 people who can regularly attend and move things forward, I don't want to make them feel like they need to get 2 more warm bodies to be performing well.

> - Four (4) are elected Contributing Seats, open to any project contributor.
>   Contributing Seats are elected every year, and serve a two-year term.  In the
>   event of a vacancy, the Steering Committee may select a replacement to serve
>   out the remainder of the term.

Reviewer 3: ```suggestion
- Four (4) are elected Contributing Seats, open to any project contributor.
  Contributing Seats serve a two-year term and are elected when their terms end.  In the
  event of a vacancy, the Steering Committee may select a replacement to serve
  out the remainder of the term.
```

Every year and 2 years seemed contradicting to me.

Reviewer 3: If you mean that e.g. 2 seats are elected this year, and 2 are elected next year, we should clarify that.

Author: Clarified, thanks!

- One (1) is an [End User Seat](#end-user-seat) appointed by the Steering
  Committee.  The End User Seat serves until the next elected Steering Committee
  appoints a replacement in the two months after members are seated, as
  described below.

> - Up to two Selected Seats (2) are appointed by the Steering Committee to

Reviewer 2: What does "up to" means exactly ? In the intro, it says, that the SC has 7 members (not: "between 5 and 7 members"). Can these selected seats also be skipped ?

Reviewer 3: Somewhere below:

> The Steering Committee is not required to fill these Selected Seats

But let me make an inline suggestion for updating the "SC has 7 members" part

Author: Oops, yes.  This is a 5-to-7 members thing -- if the SC doesn't need outside experts, they aren't required to fill these seats.

  serve out the remainder of the term.

- One (1) is an [End User Seat](#end-user-seat) appointed by the Steering
  Committee.  The End User Seat serves until the next elected Steering Committee
  appoints a replacement in the two months after members are seated, as
  described below.

- Up to two Selected Seats (2) are appointed by the Steering Committee to
  balance the set of skills on the Steering Committee.  These seats serve a
  1-year term, until Contributing Seats for the next Steering Committee election
  are seated.

Steering Committee contributing members serve for 2-year terms, staggered in
order to preserve continuity. End user and selected members serve for 1-year
terms. Every year two contributor seats are elected.

### Limitations on Company Representation

No more than two contributing seats may be held by employees of the same organization
(or conglomerate, in the case of companies owning each other). Additionally, the end
user seat must belong to an employee from a different organization. No company may
hold a majority (fifty percent or more) of the total Steering Committee seats through

> any combination of contributing, end-user, and selected seats. If the results of an

Reviewer: assuming that we're ok with the temporary KSC (the one serving until the upcoming election) not meeting this requirement

Reviewer 4: based on the text, I think the intention is more general and apply to temporary KSC as well. Maybe best to clarify this.

> any combination of contributing, end-user, and selected seats. If the results of an

Reviewer 4: ```suggestion
any combination of seats. If the results of an
```

Reviewer 4: removed the incomplete list of potential seats to clarify it relates to all types of seats.

### Limitations on Company Representation

No more than two contributing seats may be held by employees of the same organization
(or conglomerate, in the case of companies owning each other). Additionally, each end

> user seat (when available) must belong to an employee from a different

Reviewer 2: Is the end user seat optional or mandatory? Maybe we should skip "(when available)" or otherwise mention that an end user seat is optional (which then also would decrease the size of the SC)

Author: Removed, thanks!

organization. No company may hold a majority (fifty percent or more) of the total

> Steering Committee seats through any combination of contributing, end-user, and
> selected seats. If the results of an election result in greater than two employees of

Reviewer 2: ```suggestion
Steering Committee seats. If the results of an election result in greater than two employees of
```

I would just leave out the "any combination of contributing, end-user, and selected seats" as I think this is redundant and a bit confusing (what is a combination of seats, anyway?).

Author: This is just making clear that the sum total of seats can't be a majority.  This means that the following would be disallowed:

```
2 $Foo elected seats
1 $Foo end-user seat
2 Non-$Foo elected seats
1 Selected seat (non-$Foo)
```
If $Foo is an end-user company.  Another combination that would be disallowed:
```
2 $Foo elected seats
2 Non-$Foo elected seats
1 Non-$Foo end-user seat
1 $Foo selected seat
```
(Because that would give $Foo a total of 50% of the committee representation)

Author: Whoops, we actually prevent the first scenario.  But, to your point, we should move most of the company representation limits to a section that's not under "Election Procedure"

### Limitations on Company Representation

> No more than two contributing seats may be held by employees of the same organization

Reviewer: Do we need to limit the selected seats too? Otherwise one company could in theory hold 4/7 seats (two contributing seats and the two selected seats)?

Author: Good question -- this would be done with the consent of the 2 other contributing seats and the end-user seat.  I'm willing to add this restriction, but I'm also thinking of times that we've approved an exception to these rules for the TOC, so I'm wondering if we want to hem ourselves in.

Reviewer 5: Maybe just an overall limit of 3 seats per organization?  I'd love for it to be lower, but also dealing with the current reality.

Reviewer 6: I don't think we need these limits anymore and should drop them and reintroduce them later when we have more vendors actively participating in the project.

Reviewer 7: @dprotaso The problem with removing the limits is that if a malicious vendor ended up with a majority of seats (which could happen if not enough other people run), then there would be no way to reintroduce the limits because the vendor could always vote against reintroducing them, since this is the decision-making body that would make that change. 

The challenge with governance is that we need to anticipate the worst case scenarios and prevent them from happening.

Reviewer 6: @geekygirldawn yeah good point

Reviewer 5: Also, a good reason to have selected seats is to increase diversity.

Author: I think the question is, do we need to have company diversity limits on the selected seats?  I'm okay with adding that, sending a PR.

Author: Err, updating this PR.

Reviewer: I like Josh's suggestion of an overall limit

Reviewer 3: > I like Josh's suggestion of an overall limit
Same here

Author: I added an overall limit on seats.

The person filling the end user seat will take office in February.

> ## Selected Seats

Reviewer 2: This section is redundant as the definition of "selected seats" is already in the "Composition" section. I'd suggest either weaving in this information here to the definition of _selected seats_ above or having third-level sections for "### Contributing Seats", "### End User Seat" and "### Selected Seats" in the section "## Composition" above.

the person they are replacing, regardless of the length of that remainder.

In the event of a resignation or other loss of an appointed end user KSC member,
the SC will appoint a new person to fill the end user SC seat.  In the event of

> a resignation or other loss of an appointed select KSC member, the SC may choose

Reviewer 3: ```suggestion
a resignation or other loss of an appointed select SC member, the SC may choose
```

Roland, I suppose they're the same thing :)

> a resignation or other loss of an appointed select KSC member, the SC may choose

Author: Oops, we seem to have mixed the two abbreviations.  Standardized on "KSC"

| <img width="30px" src="https://github.com/pmorie.png">         | Paul Morie           | [@pmorie](https://github.com/pmorie)                 | 2020       | 2022     |
| <img width="30px" src="https://github.com/thisisnotapril.png"> | April Kyle Nassi     | [@thisisnotapril](https://github.com/thisisnotapril) | Bootstrap  | 2022     |
| <img width="30px" src="https://github.com/itsmurugappan.png">  | Murugappan Chetty    | [@itsmurugappan](https://github.com/itsmurugappan)   | 2022       | 2023     |
| <img width="30px" src="https://github.com/csantanapr.png">     | Carlos Santana       | [@csantanapr](https://github.com/csantanapr)         | 2021       | 2023     |
| <img width="30px" src="https://github.com/lance.png">          | Lance Ball           | [@lance](https://github.com/lance)                   | 2021       | 2023     |
| <img width="30px" src="https://github.com/puerco.png">         | Adolfo García Veytia | [@puerco](https://github.com/puerco)                 | 2023       | 2024     |

### Technical Oversight Committee

> 

Reviewer 2: We might want to add a section here about what the "TOC" actually _was_ as this is the first mention of the (obsolete) TOC.

Reviewer 3: Added some text here: https://github.com/knative/community/pull/1587/files#r1688479540

> To recognize the folks that have served in the TOC in the past, below we list the previous members of the TOC (sorted by their 'Term End').
> 
> | &nbsp;                                                         | Member          | Profile                                              | Term Start | Term End   |
> | -------------------------------------------------------------- | ---------------  | --------------------------------------------------- | ---------- | ---------- |
> | <img width="30px" src="https://github.com/vaikas.png">         | Ville Aikas     | [@vaikas](https://github.com/vaikas)                 | Bootstrap  | 2020       |
> | <img width="30px" src="https://github.com/tcnghia.png">        | Nghia Tran      | [@tcnghia](https://github.com/tcnghia)               | 2020       | 2021       |
> | <img width="30px" src="https://github.com/mattmoor.png">       | Matt Moore      | [@mattmoor](https://github.com/mattmoor)             | Bootstrap  | 2021       |
> | <img width="30px" src="https://github.com/grantr.png">         | Grant Rodgers   | [@grantr](https://github.com/grantr)                 | 2020       | 2021       |
> | <img width="30px" src="https://github.com/markusthoemmes.png"> | Markus Thömmes  | [@markusthoemmes](https://github.com/markusthoemmes) | 2020-06-09 | 2021-12-16 |
> | <img width="30px" src="https://github.com/julz.png">           | Julian Friedman | [@julz](https://github.com/julz)                     | 2021-05-26 | 2022-02-07 |
> | <img width="30px" src="https://github.com/rhuss.png">          | Roland Huß      | [@rhuss](https://github.com/rhuss)                   | 2021-02-16 | 2022       |
> | <img width="30px" src="https://github.com/mattmoor.png">       | Matt Moore      | [@mattmoor](https://github.com/mattmoor)             | 2022-02-07 | 2022       |
> | <img width="30px" src="https://github.com/n3wscott.png">       | Scott Nichols   | [@n3wscott](https://github.com/n3wscott)             | 2022-01-11 | 2022-09-27 |
> | <img width="30px" src="https://github.com/evankanderson.png">  | Evan Anderson   | [@evankanderson](https://github.com/evankanderson)   | Bootstrap  | 2023-06-01 |
> | <img width="30px" src="https://github.com/zroubalik.png">      | Zbynek Roubalik | [@zroubalik](https://github.com/zroubalik)           | 2022-06-01 | 2023-07-31 |
> | <img width="30px" src="https://github.com/krsna-m.png">        | Krsna Mahapatra | [@krsna-m](https://github.com/krsna-m)               | 2023-06-01 | 2024       |
> 
> 
> ---

Reviewer 3: ```suggestion
To recognize the folks that have served in the TOC* in the past, below we list the previous members of the TOC (sorted by their 'Term End').

| &nbsp;                                                         | Member          | Profile                                              | Term Start | Term End   |
| -------------------------------------------------------------- | ---------------  | --------------------------------------------------- | ---------- | ---------- |
| <img width="30px" src="https://github.com/vaikas.png">         | Ville Aikas     | [@vaikas](https://github.com/vaikas)                 | Bootstrap  | 2020       |
| <img width="30px" src="https://github.com/tcnghia.png">        | Nghia Tran      | [@tcnghia](https://github.com/tcnghia)               | 2020       | 2021       |
| <img width="30px" src="https://github.com/mattmoor.png">       | Matt Moore      | [@mattmoor](https://github.com/mattmoor)             | Bootstrap  | 2021       |
| <img width="30px" src="https://github.com/grantr.png">         | Grant Rodgers   | [@grantr](https://github.com/grantr)                 | 2020       | 2021       |
| <img width="30px" src="https://github.com/markusthoemmes.png"> | Markus Thömmes  | [@markusthoemmes](https://github.com/markusthoemmes) | 2020-06-09 | 2021-12-16 |
| <img width="30px" src="https://github.com/julz.png">           | Julian Friedman | [@julz](https://github.com/julz)                     | 2021-05-26 | 2022-02-07 |
| <img width="30px" src="https://github.com/rhuss.png">          | Roland Huß      | [@rhuss](https://github.com/rhuss)                   | 2021-02-16 | 2022       |
| <img width="30px" src="https://github.com/mattmoor.png">       | Matt Moore      | [@mattmoor](https://github.com/mattmoor)             | 2022-02-07 | 2022       |
| <img width="30px" src="https://github.com/n3wscott.png">       | Scott Nichols   | [@n3wscott](https://github.com/n3wscott)             | 2022-01-11 | 2022-09-27 |
| <img width="30px" src="https://github.com/evankanderson.png">  | Evan Anderson   | [@evankanderson](https://github.com/evankanderson)   | Bootstrap  | 2023-06-01 |
| <img width="30px" src="https://github.com/zroubalik.png">      | Zbynek Roubalik | [@zroubalik](https://github.com/zroubalik)           | 2022-06-01 | 2023-07-31 |
| <img width="30px" src="https://github.com/krsna-m.png">        | Krsna Mahapatra | [@krsna-m](https://github.com/krsna-m)               | 2023-06-01 | 2024       |

---

> * _Technical Oversight Committee (TOC)_ was a separate committee responsible for setting the overall technical direction and roadmap of the project before 2024. In 2024, SC and TOC [was merged](https://github.com/knative/community/issues/1549) into a single governance body.
```

| <img width="30px" src="https://github.com/thisisnotapril.png"> | April Kyle Nassi     | [@thisisnotapril](https://github.com/thisisnotapril) | Bootstrap  | 2022     |
| <img width="30px" src="https://github.com/itsmurugappan.png">  | Murugappan Chetty    | [@itsmurugappan](https://github.com/itsmurugappan)   | 2022       | 2023     |
| <img width="30px" src="https://github.com/csantanapr.png">     | Carlos Santana       | [@csantanapr](https://github.com/csantanapr)         | 2021       | 2023     |
| <img width="30px" src="https://github.com/lance.png">          | Lance Ball           | [@lance](https://github.com/lance)                   | 2021       | 2023     |
| <img width="30px" src="https://github.com/puerco.png">         | Adolfo García Veytia | [@puerco](https://github.com/puerco)                 | 2023       | 2024     |

### Technical Oversight Committee

To recognize the folks that have served in the TOC in the past, below we list the previous members of the TOC (sorted by their 'Term End').

The _Technical Oversight Committee (TOC)_ was a separate committee responsible
for setting the overall technical direction and roadmap of the project before

> 2024. In 2024, SC and TOC [were merged](https://github.com/knative/community/issues/1549) into a single governance body.

Reviewer: nit: personally, I would flip the order of these two paragraphs
